frontend/src/app/components/auth/auth.service.ts

```diff
@@ -155,7 +155,7 @@ export class AuthService {
   }
 
   isAuthenticated(): boolean {
-    return this._cookie.get('token') !== null;
+    return this._cookie.check('token') && this._cookie.get('token') !== null;
```

Comment:
The check for the existence of the 'token' cookie seems redundant. The `check` method already returns a boolean indicating whether the cookie exists or not, so there's no need to also use the `get` method. [medium]